File: Sommarkurser 13/C#/Assignment3/Assignment3/InputUtility.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

/// File: InputUtility.cs
/// Name: Tim Fielding
/// Date: 08/07/2013

namespace Assignment3
{
    /// <summary>
    /// Assists the input part of the program
    /// </summary>
    class InputUtility
    {
        /// <summary>
        /// Tries to parse the doubleOut parameter, returns true and changed the value of doubleOut if succeeded, otherwise not
        /// </summary>
        /// <param name="toConvert"></param>
        /// <param name="doubleOut"></param>
        /// <returns></returns>
        public static bool GetDouble(string toConvert, out double doubleOut)
        {
            return double.TryParse(toConvert, out doubleOut);
        }
        /// <summary>
        /// The methods below checks if the input is valid (non-empty string, non-zero numbers)
        /// </summary>
        /// <param name="toCheck"></param>
        /// <returns></returns>
        public static bool ValidateString(string toCheck)
        {
            return !string.IsNullOrEmpty(toCheck);
        }

        public static bool ValidateDouble(double number)
        {
            return !(number >= 0.0d);
        }
    }
}

File: Sommarkurser 13/Visual Basic.NET/Assignment3/Assignment3/InputUtility.vb
Option Explicit On
Option Strict On

'InputUtility.vb
'Created by: Tim Fielding 08/07/2013
'Revised:

''' <summary>
''' Class used to check if inputs from the user checks out and to check/"convert" some types to others
''' </summary>
''' <remarks></remarks>
Public Class InputUtility
    ''' <summary>
    ''' Takes a string and a double as parameters, checks with the tryparse function if it works to parse the double,
    ''' if it does the double is set to the double-value of the string
    ''' </summary>
    ''' <param name="toConvert"></param>
    ''' <param name="doubleOut"></param>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Public Shared Function GetDouble(ByVal toConvert As String, ByRef doubleOut As Double) As Boolean
        Return Double.TryParse(toConvert, doubleOut)
    End Function
    ''' <summary>
    ''' Returns false if a string is empty, true if it's not
    ''' </summary>
    ''' <param name="toCheck"></param>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Public Shared Function ValidateString(ByVal toCheck As String) As Boolean
        Return Not String.IsNullOrEmpty(toCheck)
    End Function
    ''' <summary>
    ''' Returns true if a double is more than 0, returns false otherwise
    ''' </summary>
    ''' <param name="toCheck"></param>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Public Shared Function ValidateDouble(ByVal toCheck As Double) As Boolean
        Return Not (toCheck > 0D)
    End Function

End Class